File: app/src/main/java/xyz/farshad/vocab/component/Server/RemoteServer.java
package xyz.farshad.vocab.component.Server;

import android.util.Log;

import org.json.JSONObject;

public class RemoteServer {
    private RemoteGet remoteGet;

    public RemoteServer() {
        remoteGet = new RemoteGet();
    }

    public JSONObject get(String link){

        try {
            return remoteGet.send(link, null);
        } catch (Exception e) {
            Log.i("WEB", e.getMessage());
        }
        return null;
    }
}

File: app/src/main/java/xyz/farshad/vocab/component/Server/RemoteServer.kt
package xyz.farshad.vocab.component.Server

import android.util.Log
import org.json.JSONObject

/**
 * this class is a facade for RemotePost and RemoteGet classes
 */
class RemoteServer {
    private val remoteGet: RemoteGet

    init {
        remoteGet = RemoteGet()
    }

    /**
     * call remoteGet class and send request
     *
     * @param link
     * @return JSONObject
     */
    operator fun get(link: String): Any? {

        try {
            return remoteGet.send(link, null)
        } catch (e: Exception) {
            return e
        }

    }
}
